server/sonar-server/src/main/java/org/sonar/server/issue/ws/SearchAction.java

```diff
@@ -359,13 +369,43 @@ private SearchWsResponse doHandle(SearchRequest request, Request wsRequest) {
     // Must be done after loading of data as the "hidden" facet "debt"
     // can be used to get total debt.
     facets = reorderFacets(facets, options.getFacets());
+    replaceRuleIdsByRuleKeys(facets);
 
     // FIXME allow long in Paging
     Paging paging = forPageIndex(options.getPage()).withPageSize(options.getLimit()).andTotal((int) result.getHits().getTotalHits());
 
     return searchResponseFormat.formatSearch(additionalFields, data, paging, facets);
   }
 
+  private void replaceRuleIdsByRuleKeys(@Nullable Facets facets) {
+    if (facets == null || facets.get(PARAM_RULES) == null) {
+      return;
+    }
+    // The facet for PARAM_RULES contains the id of the rule as the key
+    // We need to update the key to be a RuleKey
+    LinkedHashMap<String, Long> rulesFacet = facets.get(PARAM_RULES);
+
+    try (DbSession dbSession = dbClient.openSession(false)) {
+      Map<Integer, RuleKey> idToRuleKey = dbClient.ruleDao()
+        .selectDefinitionByIds(dbSession, Collections2.transform(rulesFacet.keySet(), Integer::parseInt))
+        .stream()
+        .collect(Collectors.toMap(RuleDefinitionDto::getId, RuleDefinitionDto::getKey));
+
+      LinkedHashMap<String, Long> newRulesFacet = new LinkedHashMap<>();
+      rulesFacet.forEach((k, v) -> {
+        RuleKey ruleKey = idToRuleKey.get(Integer.parseInt(k));
+        if (ruleKey != null) {
+          newRulesFacet.put(ruleKey.toString(), v);
+        } else {
+          // RuleKey not found ES/DB incorrect?
+          LOGGER.error("Rule with id {} is not available in database", k);
+        }
+      });
+      rulesFacet.clear();
+      rulesFacet.putAll(newRulesFacet);
+    }
+  }
+
   private static SearchOptions createSearchOptionsFromRequest(SearchRequest request) {
     SearchOptions options = new SearchOptions();
     options.setPage(request.getPage(), request.getPageSize());
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: "setPage" is marked "javax.annotation.ParametersAreNonnullByDefault" but parameter 2 could be null. ([squid:S2637](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2637))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Feh%2FSONAR-10303&issues=AWEy-NAVnGZqNhaYO-6T&open=AWEy-NAVnGZqNhaYO-6T)

[ISSUE_KEY]: AWEy-NAVnGZqNhaYO-6T